fix(2021.2): fix most 'target of xref not found' errors

tbouffard opened the pull request:
This continues the work started in #1819

**Done**
- fix xref to existing pages that used wrong page ids
- the fixed xrefs now use direct page id coordinates instead of relying on alias to prepare the future (don't rely on old page name and location out of specific module)
- a lot of pages are updated but a lot of xrefs were broken


**Remaining errors**
Some pages have been removed, content has been moved and no alias exist, so some xref can not easily fixed.
They must be replaced by xref to existing pages or completely removed. This is out of the scope of this PR (see with Product Team). Some issues are coming from version 2021.1.

<details>
<summary>Remaining xref errors due to missing pages in commit 6af3263c </summary>
[18:03:31.879] ERROR (asciidoctor): target of xref not found: ROOT:xx.adoc
    file: /development/bonita/docs/bonita-doc/modules/ROOT/pages/deployment-index.adoc:10
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[18:03:31.882] ERROR (asciidoctor): target of xref not found: ROOT:xxg.adoc
    file: /development/bonita/docs/bonita-doc/modules/ROOT/pages/deployment-index.adoc:15
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[18:03:31.883] ERROR (asciidoctor): target of xref not found: ROOT:xx.adoc
    file: /development/bonita/docs/bonita-doc/modules/ROOT/pages/deployment-index.adoc:20
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[18:03:31.894] ERROR (asciidoctor): target of xref not found: ROOT:xx.adoc
    file: /development/bonita/docs/bonita-doc/modules/ROOT/pages/development-resources-index.adoc:10
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[18:03:31.895] ERROR (asciidoctor): target of xref not found: ROOT:xx.adoc
    file: /development/bonita/docs/bonita-doc/modules/ROOT/pages/development-resources-index.adoc:15
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[18:03:31.898] ERROR (asciidoctor): target of xref not found: ROOT:xx.adoc
    file: /development/bonita/docs/bonita-doc/modules/ROOT/pages/development-resources-index.adoc:20
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[18:03:34.989] ERROR (asciidoctor): target of xref not found: multi-tenancy-and-tenant-configuration.adoc
    file: /development/bonita/docs/bonita-doc/modules/bonita-overview/pages/bonita-bpm-overview.adoc:71
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[18:03:35.026] ERROR (asciidoctor): target of xref not found: ROOT:managing_extension_studio.adoc
    file: /development/bonita/docs/bonita-doc/modules/bonita-overview/pages/bonita-studio.adoc:23
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[18:03:35.083] ERROR (asciidoctor): target of xref not found: ROOT:managing_extension_studio.adoc
    file: /development/bonita/docs/bonita-doc/modules/bonita-overview/pages/project-structure.adoc:143
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[18:03:36.164] ERROR (asciidoctor): target of xref not found: ROOT:custom-profiles.adoc
    file: /development/bonita/docs/bonita-doc/modules/identity/pages/rest-api-authorization.adoc:323
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[18:03:36.266] ERROR (asciidoctor): target of xref not found: multi-tenancy-and-tenant-configuration.adoc
    file: /development/bonita/docs/bonita-doc/modules/identity/pages/single-sign-on-with-kerberos.adoc:210
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[18:03:37.118] ERROR (asciidoctor): target of xref not found: ROOT:managing_extension_studio.adoc
    file: /development/bonita/docs/bonita-doc/modules/process/pages/actors.adoc:61
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[18:03:37.174] ERROR (asciidoctor): target of xref not found: ROOT:organization-in-bonita-applications-overview.adoc
    file: /development/bonita/docs/bonita-doc/modules/process/pages/configure-a-test-organization.adoc:71
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[18:03:37.204] ERROR (asciidoctor): target of xref not found: ROOT:managing_extension_studio.adoc
    file: /development/bonita/docs/bonita-doc/modules/process/pages/connectivity-overview.adoc:28
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[18:03:37.509] ERROR (asciidoctor): target of xref not found: ROOT:bonita-bpm-studio-hints-and-tips.adoc
    file: /development/bonita/docs/bonita-doc/modules/process/pages/diagram-overview.adoc:7
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[18:03:37.510] ERROR (asciidoctor): target of xref not found: bonita-bpm-studio-hints-and-tips.adoc
    file: /development/bonita/docs/bonita-doc/modules/process/pages/diagram-overview.adoc:7
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[18:03:38.301] ERROR (asciidoctor): target of xref not found: ROOT:manage-dependencies.adoc
    file: /development/bonita/docs/bonita-doc/modules/process/pages/web-service-connector-overview.adoc:23
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[18:03:38.620] ERROR (asciidoctor): target of xref not found: ROOT:organization-in-bonita-applications-overview.adoc
    file: /development/bonita/docs/bonita-doc/modules/runtime/pages/applications.adoc:28
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[18:03:39.447] ERROR (asciidoctor): target of xref not found: ROOT:process-manager.adoc
    file: /development/bonita/docs/bonita-doc/modules/runtime/pages/monitoring.adoc:29
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[18:03:39.609] ERROR (asciidoctor): target of xref not found: ROOT:organization-in-bonita-applications-overview.adoc
    file: /development/bonita/docs/bonita-doc/modules/runtime/pages/profiles-applications-overview.adoc:39
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[18:03:40.224] ERROR (asciidoctor): target of xref not found: ROOT:managing_extension_studio.adoc
    file: /development/bonita/docs/bonita-doc/modules/software-extensibility/pages/software-extensibility.adoc:26
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[18:03:40.225] ERROR (asciidoctor): target of xref not found: ROOT:managing_extension_studio.adoc
    file: /development/bonita/docs/bonita-doc/modules/software-extensibility/pages/software-extensibility.adoc:34
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[18:03:40.227] ERROR (asciidoctor): target of xref not found: ROOT:managing_extension_studio.adoc
    file: /development/bonita/docs/bonita-doc/modules/software-extensibility/pages/software-extensibility.adoc:55
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
</details>


<details>
<summary>Examples of Antora 3 errors due to wrong xref (a lot of fixes are already applied, commit 4b95fb3)</summary>
[15:42:53.545] ERROR (asciidoctor): target of xref not found: ROOT:profiles.adoc
    file: /development/bonita/docs/bonita-doc/modules/applications/pages/custom-applications-index.adoc:39
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:53.556] ERROR (asciidoctor): target of xref not found: ROOT:applications-deep-dive-index.adoc
    file: /development/bonita/docs/bonita-doc/modules/applications/pages/custom-applications-index.adoc:99
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:53.717] ERROR (asciidoctor): target of xref not found: design-application-page.adoc
    file: /development/bonita/docs/bonita-doc/modules/applications/pages/multi-language-pages.adoc:52
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:53.723] ERROR (asciidoctor): target of xref not found: applications.adoc
    file: /development/bonita/docs/bonita-doc/modules/applications/pages/multi-language-pages.adoc:159
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:53.723] ERROR (asciidoctor): target of xref not found: create-application.adoc
    file: /development/bonita/docs/bonita-doc/modules/applications/pages/multi-language-pages.adoc:159
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:53.806] ERROR (asciidoctor): target of xref not found: live-update.adoc
    file: /development/bonita/docs/bonita-doc/modules/bonita-overview/pages/bonita-bpm-overview.adoc:71
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:53.806] ERROR (asciidoctor): target of xref not found: multi-tenancy-and-tenant-configuration.adoc
    file: /development/bonita/docs/bonita-doc/modules/bonita-overview/pages/bonita-bpm-overview.adoc:71
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:53.825] ERROR (asciidoctor): target of xref not found: user-application-overview.adoc
    file: /development/bonita/docs/bonita-doc/modules/bonita-overview/pages/bonita-purpose.adoc:41
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:53.848] ERROR (asciidoctor): target of xref not found: contracts-and-contexts.adoc
    file: /development/bonita/docs/bonita-doc/modules/bonita-overview/pages/bonita-studio.adoc:21
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:53.849] ERROR (asciidoctor): target of xref not found: actors.adoc
    file: /development/bonita/docs/bonita-doc/modules/bonita-overview/pages/bonita-studio.adoc:22
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:53.850] ERROR (asciidoctor): target of xref not found: ROOT:managing_extension_studio.adoc
    file: /development/bonita/docs/bonita-doc/modules/bonita-overview/pages/bonita-studio.adoc:23
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:53.906] ERROR (asciidoctor): target of xref not found: ROOT:creating-an-actor-filter.adoc
    file: /development/bonita/docs/bonita-doc/modules/bonita-overview/pages/project-structure.adoc:37
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:53.910] ERROR (asciidoctor): target of xref not found: ROOT:applicationcreation.adoc
    file: /development/bonita/docs/bonita-doc/modules/bonita-overview/pages/project-structure.adoc:82
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:53.916] ERROR (asciidoctor): target of xref not found: ROOT:managing_extension_studio.adoc
    file: /development/bonita/docs/bonita-doc/modules/bonita-overview/pages/project-structure.adoc:143
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:53.963] ERROR (asciidoctor): target of xref not found: custom-widgets.adoc
    file: /development/bonita/docs/bonita-doc/modules/bonita-overview/pages/ui-designer-overview.adoc:34
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:53.987] ERROR (asciidoctor): target of xref not found: diagram-overview.adoc
    file: /development/bonita/docs/bonita-doc/modules/bonita-overview/pages/using-Bonita-Studio.adoc:40
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:55.311] ERROR (asciidoctor): target of xref not found: bonita-platform-setup.adoc#update_platform_conf
    file: /development/bonita/docs/bonita-doc/modules/api/pages/manage-the-platform.adoc:46
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:56.337] ERROR (asciidoctor): target of xref not found: ROOT:xx.adoc
    file: /development/bonita/docs/bonita-doc/modules/ROOT/pages/deployment-index.adoc:10
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:56.338] ERROR (asciidoctor): target of xref not found: ROOT:xxg.adoc
    file: /development/bonita/docs/bonita-doc/modules/ROOT/pages/deployment-index.adoc:15
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:56.339] ERROR (asciidoctor): target of xref not found: ROOT:xx.adoc
    file: /development/bonita/docs/bonita-doc/modules/ROOT/pages/deployment-index.adoc:20
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:56.348] ERROR (asciidoctor): target of xref not found: ROOT:xx.adoc
    file: /development/bonita/docs/bonita-doc/modules/ROOT/pages/development-resources-index.adoc:10
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:56.349] ERROR (asciidoctor): target of xref not found: ROOT:xx.adoc
    file: /development/bonita/docs/bonita-doc/modules/ROOT/pages/development-resources-index.adoc:15
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:56.350] ERROR (asciidoctor): target of xref not found: ROOT:xx.adoc
    file: /development/bonita/docs/bonita-doc/modules/ROOT/pages/development-resources-index.adoc:20
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:56.923] ERROR (asciidoctor): target of xref not found: ROOT:what-is-bonita.adoc
    file: /development/bonita/docs/bonita-doc/modules/data/pages/list-of-documents.adoc:16
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:56.939] ERROR (asciidoctor): target of xref not found: ROOT:process-configuration-overview.adoc
    file: /development/bonita/docs/bonita-doc/modules/data/pages/parameters.adoc:31
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:56.999] ERROR (asciidoctor): target of xref not found: customize-living-application-theme.adoc
    file: /development/bonita/docs/bonita-doc/modules/getting-started/pages/create-application.adoc:41
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:57.076] ERROR (asciidoctor): target of xref not found: organization-overview.adoc
    file: /development/bonita/docs/bonita-doc/modules/getting-started/pages/define-who-can-do-what.adoc:7
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:57.099] ERROR (asciidoctor): target of xref not found: variables.adoc
    file: /development/bonita/docs/bonita-doc/modules/getting-started/pages/design-application-page.adoc:27
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:57.235] ERROR (asciidoctor): target of xref not found: admin-application-users-list.adoc
    file: /development/bonita/docs/bonita-doc/modules/identity/pages/approaches-to-managing-organizations-and-actor-mapping.adoc:30
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:57.236] ERROR (asciidoctor): target of xref not found: import-export-an-organization.adoc
    file: /development/bonita/docs/bonita-doc/modules/identity/pages/approaches-to-managing-organizations-and-actor-mapping.adoc:34
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:57.354] ERROR (asciidoctor): target of xref not found: ROOT:page-and-form-development-overview.adoc
    file: /development/bonita/docs/bonita-doc/modules/identity/pages/custom-authorization-rule-mapping.adoc:13
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:57.562] ERROR (asciidoctor): target of xref not found: admin-application-roles-list.adoc
    file: /development/bonita/docs/bonita-doc/modules/identity/pages/organization-management-in-bonita-bpm-studio.adoc:62
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:57.563] ERROR (asciidoctor): target of xref not found: admin-application-users-list.adoc
    file: /development/bonita/docs/bonita-doc/modules/identity/pages/organization-management-in-bonita-bpm-studio.adoc:62
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:57.658] ERROR (asciidoctor): target of xref not found: ROOT:custom-profiles.adoc
    file: /development/bonita/docs/bonita-doc/modules/identity/pages/rest-api-authorization.adoc:323
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:57.756] ERROR (asciidoctor): target of xref not found: tomcat-bundle.adoc#configuration
    file: /development/bonita/docs/bonita-doc/modules/identity/pages/single-sign-on-with-kerberos.adoc:137
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:57.759] ERROR (asciidoctor): target of xref not found: multi-tenancy-and-tenant-configuration.adoc
    file: /development/bonita/docs/bonita-doc/modules/identity/pages/single-sign-on-with-kerberos.adoc:210
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:57.814] ERROR (asciidoctor): target of xref not found: enable-cors-in-tomcat-bundle.adoc#_add_cors_filter
    file: /development/bonita/docs/bonita-doc/modules/identity/pages/single-sign-on-with-oidc.adoc:405
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:57.815] ERROR (asciidoctor): target of xref not found: enable-cors-in-tomcat-bundle.adoc#_choose_cookies_samesite_policy
    file: /development/bonita/docs/bonita-doc/modules/identity/pages/single-sign-on-with-oidc.adoc:405
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:57.961] ERROR (asciidoctor): target of xref not found: documents-index.adoc
    file: /development/bonita/docs/bonita-doc/modules/best-practices/pages/design-methodology.adoc:88
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:57.962] ERROR (asciidoctor): target of xref not found: organization-overview.adoc
    file: /development/bonita/docs/bonita-doc/modules/best-practices/pages/design-methodology.adoc:90
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:57.962] ERROR (asciidoctor): target of xref not found: software-extensibility.adoc
    file: /development/bonita/docs/bonita-doc/modules/best-practices/pages/design-methodology.adoc:90
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:57.965] ERROR (asciidoctor): target of xref not found: bonita-layout.adoc
    file: /development/bonita/docs/bonita-doc/modules/best-practices/pages/design-methodology.adoc:129
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:57.967] ERROR (asciidoctor): target of xref not found: software-extensibility.adoc
    file: /development/bonita/docs/bonita-doc/modules/best-practices/pages/design-methodology.adoc:176
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:57.968] ERROR (asciidoctor): target of xref not found: ROOT:process-configuration-overview.adoc
    file: /development/bonita/docs/bonita-doc/modules/best-practices/pages/design-methodology.adoc:184
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:57.970] ERROR (asciidoctor): target of xref not found: ROOT:process-configuration-overview.adoc
    file: /development/bonita/docs/bonita-doc/modules/best-practices/pages/design-methodology.adoc:189
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:57.971] ERROR (asciidoctor): target of xref not found: bdm-access-control.adoc
    file: /development/bonita/docs/bonita-doc/modules/best-practices/pages/design-methodology.adoc:199
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:57.971] ERROR (asciidoctor): target of xref not found: actor-filtering.adoc
    file: /development/bonita/docs/bonita-doc/modules/best-practices/pages/design-methodology.adoc:199
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:57.972] ERROR (asciidoctor): target of xref not found: actors.adoc
    file: /development/bonita/docs/bonita-doc/modules/best-practices/pages/design-methodology.adoc:199
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:57.972] ERROR (asciidoctor): target of xref not found: profiles-overview.adoc
    file: /development/bonita/docs/bonita-doc/modules/best-practices/pages/design-methodology.adoc:199
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:58.082] ERROR (asciidoctor): target of xref not found: software-extensibility.adoc
    file: /development/bonita/docs/bonita-doc/modules/pages-and-forms/pages/create-or-modify-a-page.adoc:19
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:58.182] ERROR (asciidoctor): target of xref not found: live-update.adoc#form-mapping
    file: /development/bonita/docs/bonita-doc/modules/pages-and-forms/pages/forms.adoc:36
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:58.268] ERROR (asciidoctor): target of xref not found: ROOT::rta-mail-template.adoc
    file: /development/bonita/docs/bonita-doc/modules/pages-and-forms/pages/pages-and-forms.adoc:87
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:58.300] ERROR (asciidoctor): target of xref not found: ROOT:resource-management.adoc]#modify
    file: /development/bonita/docs/bonita-doc/modules/pages-and-forms/pages/pages.adoc:68
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:58.301] ERROR (asciidoctor): target of xref not found: ROOT:applications.adoc]#specify-pages
    file: /development/bonita/docs/bonita-doc/modules/pages-and-forms/pages/pages.adoc:72
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:58.301] ERROR (asciidoctor): target of xref not found: applications.adoc]]#define-navigation
    file: /development/bonita/docs/bonita-doc/modules/pages-and-forms/pages/pages.adoc:72
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:58.530] ERROR (asciidoctor): target of xref not found: ROOT:managing_extension_studio.adoc
    file: /development/bonita/docs/bonita-doc/modules/process/pages/actors.adoc:61
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:58.573] ERROR (asciidoctor): target of xref not found: ROOT:organization-in-bonita-applications-overview.adoc
    file: /development/bonita/docs/bonita-doc/modules/process/pages/configure-a-test-organization.adoc:71
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:58.592] ERROR (asciidoctor): target of xref not found: ROOT:managing_extension_studio.adoc
    file: /development/bonita/docs/bonita-doc/modules/process/pages/connectivity-overview.adoc:28
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:58.640] ERROR (asciidoctor): target of xref not found: product-versioning.adoc#_technical_id
    file: /development/bonita/docs/bonita-doc/modules/process/pages/connector-archetype-tutorial.adoc:39
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:58.653] ERROR (asciidoctor): target of xref not found: configure-maven.adoc#repositories
    file: /development/bonita/docs/bonita-doc/modules/process/pages/connector-archetype-tutorial.adoc:708
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:58.654] ERROR (asciidoctor): target of xref not found: configure-maven.adoc#servers
    file: /development/bonita/docs/bonita-doc/modules/process/pages/connector-archetype-tutorial.adoc:723
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:58.655] ERROR (asciidoctor): target of xref not found: configure-maven.adoc#encryption
    file: /development/bonita/docs/bonita-doc/modules/process/pages/connector-archetype-tutorial.adoc:729
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:58.684] ERROR (asciidoctor): target of xref not found: product-versioning.adoc#_technical_id
    file: /development/bonita/docs/bonita-doc/modules/process/pages/connector-archetype.adoc:39
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:58.780] ERROR (asciidoctor): target of xref not found: ROOT:bonita-bpm-studio-hints-and-tips.adoc
    file: /development/bonita/docs/bonita-doc/modules/process/pages/diagram-overview.adoc:7
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:58.780] ERROR (asciidoctor): target of xref not found: bonita-bpm-studio-hints-and-tips.adoc
    file: /development/bonita/docs/bonita-doc/modules/process/pages/diagram-overview.adoc:7
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:59.190] ERROR (asciidoctor): target of xref not found: ROOT:search-index.adoc
    file: /development/bonita/docs/bonita-doc/modules/process/pages/pools-and-lanes.adoc:22
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:59.345] ERROR (asciidoctor): target of xref not found: ROOT:web-service-tutorial.adoc
    file: /development/bonita/docs/bonita-doc/modules/process/pages/web-service-connector-overview.adoc:19
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:59.346] ERROR (asciidoctor): target of xref not found: ROOT:manage-dependencies.adoc
    file: /development/bonita/docs/bonita-doc/modules/process/pages/web-service-connector-overview.adoc:23
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:59.582] ERROR (asciidoctor): target of xref not found: customize-living-application-theme.adoc
    file: /development/bonita/docs/bonita-doc/modules/runtime/pages/application-directory.adoc:66
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:59.606] ERROR (asciidoctor): target of xref not found: forms.adoc
    file: /development/bonita/docs/bonita-doc/modules/runtime/pages/applications.adoc:16
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:59.606] ERROR (asciidoctor): target of xref not found: forms.adoc
    file: /development/bonita/docs/bonita-doc/modules/runtime/pages/applications.adoc:17
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:59.608] ERROR (asciidoctor): target of xref not found: ROOT:organization-in-bonita-applications-overview.adoc
    file: /development/bonita/docs/bonita-doc/modules/runtime/pages/applications.adoc:28
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:59.610] ERROR (asciidoctor): target of xref not found: ui-designer-overview.adoc
    file: /development/bonita/docs/bonita-doc/modules/runtime/pages/applications.adoc:90
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:59.615] ERROR (asciidoctor): target of xref not found: ROOT:cache-configuration-and-policy.adoc#applications
    file: /development/bonita/docs/bonita-doc/modules/runtime/pages/applications.adoc:207
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:59.662] ERROR (asciidoctor): target of xref not found: ROOT:cache-configuration-and-policy.adoc
    file: /development/bonita/docs/bonita-doc/modules/runtime/pages/bonita-applications-deep-dive.adoc:18
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:42:59.880] ERROR (asciidoctor): target of xref not found: custom-user-information-in-bonita-bpm-studio.adoc
    file: /development/bonita/docs/bonita-doc/modules/runtime/pages/custom-user-information-in-bonita-applications.adoc:5
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:43:00.097] ERROR (asciidoctor): target of xref not found: event-handlers.adoc
    file: /development/bonita/docs/bonita-doc/modules/runtime/pages/execution-sequence-states-and-transactions.adoc:125
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:43:00.137] ERROR (asciidoctor): target of xref not found: user-authentication-overview.adoc
    file: /development/bonita/docs/bonita-doc/modules/runtime/pages/first-steps-after-setup.adoc:94
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:43:00.300] ERROR (asciidoctor): target of xref not found: ROOT:cache-configuration-and-policy.adoc
    file: /development/bonita/docs/bonita-doc/modules/runtime/pages/live-update.adoc:93
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:43:00.329] ERROR (asciidoctor): target of xref not found: ROOT:process-manager.adoc
    file: /development/bonita/docs/bonita-doc/modules/runtime/pages/monitoring.adoc:29
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:43:00.475] ERROR (asciidoctor): target of xref not found: ROOT:organization-in-bonita-applications-overview.adoc
    file: /development/bonita/docs/bonita-doc/modules/runtime/pages/profiles-applications-overview.adoc:39
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:43:00.539] ERROR (asciidoctor): target of xref not found: runtime-administration-index.adoc
    file: /development/bonita/docs/bonita-doc/modules/runtime/pages/runtime-index.adoc:32
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:43:00.541] ERROR (asciidoctor): target of xref not found: migrate-from-an-earlier-version-of-bonita-bpm.adoc
    file: /development/bonita/docs/bonita-doc/modules/runtime/pages/runtime-index.adoc:45
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:43:00.624] ERROR (asciidoctor): target of xref not found: customize-living-application-theme.adoc
    file: /development/bonita/docs/bonita-doc/modules/runtime/pages/super-administrator-application-overview.adoc:74
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:43:00.943] WARN (asciidoctor): id assigned to section already in use: git-troubleshooting
    file: /development/bonita/docs/bonita-doc/modules/setup-dev-environment/pages/workspaces-and-repositories.adoc:316
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:43:01.038] ERROR (asciidoctor): target of xref not found: ROOT:managing_extension_studio.adoc
    file: /development/bonita/docs/bonita-doc/modules/software-extensibility/pages/software-extensibility.adoc:26
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:43:01.039] ERROR (asciidoctor): target of xref not found: ROOT:managing_extension_studio.adoc
    file: /development/bonita/docs/bonita-doc/modules/software-extensibility/pages/software-extensibility.adoc:34
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:43:01.040] ERROR (asciidoctor): target of xref not found: ROOT:managing_extension_studio.adoc
    file: /development/bonita/docs/bonita-doc/modules/software-extensibility/pages/software-extensibility.adoc:55
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
[15:43:01.041] ERROR (asciidoctor): target of xref not found: custom-widgets.adoc
    file: /development/bonita/docs/bonita-doc/modules/software-extensibility/pages/software-extensibility.adoc:68
    source: /development/bonita/docs/bonita-doc (refname: 2021.2_links_fixes_and_improvements <worktree>)
</details>

tbouffard committed:
fix xref

tbouffard committed:
fix xref (module ref)

tbouffard committed:
fix xref (remove extra ROOT module)

tbouffard committed:
fix xref (module ref)

tbouffard committed:
fix xref (module ref)

tbouffard committed:
fix xref (module ref)

tbouffard committed:
fix xref (module ref)

tbouffard committed:
fix xref (module ref)

tbouffard committed:
fix xref (module ref)

tbouffard committed:
fix xref (module ref)

tbouffard committed:
fix xref (module ref)

tbouffard committed:
fix xref (module ref)

tbouffard committed:
fix xref (module ref)

tbouffard committed:
fix xref (module ref)

tbouffard committed:
fix xref (module ref) + use anchor

tbouffard committed:
RESTORE content change only

This reverts commit a51db14bed9c393ba40f437df61f37cfde52da36.

tbouffard committed:
fix xref

tbouffard committed:
fix xref (including to software-extensibility.adoc)

tbouffard committed:
fix xref to applications:layout-development.adoc

tbouffard committed:
fix xref to getting-started:design-application-page.adoc

tbouffard committed:
fix xref to identity:organization-overview.adoc

tbouffard committed:
fix xref to applications:bonita-layout.adoc

tbouffard committed:
fix xref to process:process-configuration-index.adoc

tbouffard committed:
fix xref to identity:bdm-access-control.adoc

tbouffard committed:
fix xref to process:actor-filtering.adoc

tbouffard committed:
fix xref to process:actors.adoc

tbouffard committed:
fix xref to identity:profiles-overview.adoc

tbouffard committed:
fix xref to identity:rest-api-authorization.adoc

tbouffard committed:
fix xref to identity:profile-creation.adoc

tbouffard committed:
fix xref to runtime:cache-configuration-and-policy.adoc

tbouffard committed:
fix xref to identity:user-authentication-overview.adoc

tbouffard committed:
fix xref to integration:event-handlers.adoc

tbouffard committed:
fix xref to identity:custom-user-information-in-bonita-bpm-studio.adoc

tbouffard committed:
fix xref to bonita-overview:ui-designer-overview.adoc

tbouffard committed:
fix xref to pages-and-forms:forms.adoc

tbouffard committed:
fix xref to process:web-service-tutorial.adoc

tbouffard committed:
fix xref to data:define-a-search-index.adoc (+ add alias)

tbouffard committed:
fix xref to version-update:product-versioning.adoc

tbouffard committed:
fix xref to setup-dev-environment:configure-maven.adoc

tbouffard committed:
fix xref to runtime:applications.adoc

tbouffard committed:
fix xref to rta-mail-template.adoc

tbouffard committed:
fix xref to runtime:live-update.adoc

tbouffard committed:
fix xref to software-extensibility:software-extensibility.adoc

tbouffard committed:
fix xref to security:enable-cors-in-tomcat-bundle.adoc

tbouffard committed:
fix xref to runtime:tomcat-bundle.adoc

tbouffard committed:
fix xref to runtime:admin-application-groups-list.adoc

tbouffard committed:
fix xref to runtime:admin-application-roles-list.adoc

tbouffard committed:
fix xref to runtime:admin-application-users-list.adoc

tbouffard committed:
fix xref to runtime:admin-application-process-list.adoc

tbouffard committed:
fix xref to data:contracts-and-contexts.adoc

tbouffard committed:
fix xref to data:data-management.adoc

tbouffard committed:
fix xref to pages-and-forms:variables.adoc

tbouffard committed:
fix xref to bonita-overview:what-is-bonita-index.adoc

tbouffard committed:
fix xref to process:diagram-overview.adoc

tbouffard committed:
fix xref to applications:application-creation.adoc

tbouffard committed:
fix wrong xref to security:enable-cors-in-tomcat-bundle.adoc

tbouffard committed:
fix xref to runtime:import-export-an-organization.adoc

tbouffard committed:
fix xref to runtime:applications.adoc

tbouffard committed:
fix xref to version-update:product-versioning.adoc

tbouffard committed:
fix xref to runtime administration-index.adoc

tbouffard committed:
fix xref to runtime:user-application-overview.adoc

tbouffard committed:
fix xref to runtime:runtime-installation-index.adoc

tbouffard committed:
fix xref to runtime:applications.adoc

tbouffard committed:
fix xref to pages-and-forms:pages-and-forms.adoc

tbouffard committed:
fix xref to process:actor-filtering.adoc

alachambre commented:
So you basically edited all the links in the doc to remove the ROOT module hack? 😮 
I didn't have the courage to do it, thanks ^^

I'll not test them all, I guess you are confident?

tbouffard commented:
> So you basically edited all the links in the doc to remove the ROOT module hack? open_mouth

No, I only changed broken xrefs and xrefs associated to the same target pages (I remove the ROOT usage at that time).
Removing all ROOT references can be done later. I initially planned to do it as part of this PR, but this would have been too long.

To detect broken links, I have generated the preview with Antora 3 beta-1 (and experimental work available in https://github.com/bonitasoft/bonita-documentation-site/pull/266) which provides good feedback thanks to its new logging system (including file and line about the error location). You can find the remaining xref errors in the PR description.

> I'll not test them all, I guess you are confident?

So, yes I am confident, and I didn't check all resulting fixed links. Even if it is not perfect and errors remains, the situation is better than before, and we will be able to improve it later.

alachambre commented:
I'm not sure to understand, you updated some xref that were fine with the ROOT alias (for ex, the first one in api-extensions.adoc). 
But whatever, the result looks good 👍

tbouffard commented:
> I'm not sure to understand, you updated some xref that were fine with the ROOT alias (for ex, the first one in api-extensions.adoc). But whatever, the result looks good +1

Yes I updated some working xrefs using the ROOT alias like the one you are mentioning.

Here is the process I mainly followed in this Pull Request. For a given broken xref, I searched all xrefs related to the target page and updated them all to use the non alias xref.
For instance, there were some `xref:process:actors.adoc` that were wrong (like in `modules/bonita-overview/pages/bonita-studio.adoc`), in that case, I changed **all** xref related to the `actors.adoc` page, using the right module name and keeping the non-module xref form when the xrefs were used in the same module as the target page.

alachambre commented:
Okay got it!
